#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.0019494996755559078 -0.7557129386832103 -0.6549001097555271 3.1463926687888857
  position 3.006209121756935 6.8176443895636565 1.940770327555809
}
Robot {
  translation 0.8469159403013049 -2.0551994631798527 3.2127686791317247
  rotation -0.0001399200875463347 0.9999996671282383 0.0008038443765643848 2.4845915296721692
  children [
    InertialUnit {
      children [
        Solid {
          translation -0.06 0.25 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0.2 1
              }
              geometry Box {
                size 0.05 0.01 0.15
              }
            }
          ]
          boundingObject Box {
            size 0.05 0.01 0.15
          }
          physics Physics {
          }
        }
      ]
      boundingObject Box {
        size 0.1 0.1 0.1
      }
      physics Physics {
      }
    }
    Display {
      translation -0.01 0.25 0
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.05 0.01 0.15
          }
        }
      ]
      boundingObject Box {
        size 0.05 0.01 0.15
      }
      physics Physics {
      }
    }
    GPS {
      children [
        Solid {
          translation 0.04 0.25 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
              }
              geometry Box {
                size 0.05 0.01 0.15
              }
            }
          ]
          boundingObject Box {
            size 0.05 0.01 0.15
          }
          physics Physics {
          }
        }
      ]
      boundingObject Box {
        size 0.1 0.1 0.1
      }
      physics Physics {
      }
    }
    RobotisLds01 {
      translation 0 0.27 0
    }
    DEF ds_right DistanceSensor {
      translation 0.175 0.1 0.1
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.05 0.05 0.05
          }
        }
      ]
      name "ds_right"
    }
    DEF ds_left DistanceSensor {
      translation 0.175 0.1 -0.1
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.05 0.05 0.05
          }
        }
      ]
      name "ds_left"
    }
    Solid {
      translation 0.14 -0.25 0
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Sphere {
            radius 0.05
          }
        }
      ]
      name "solid(3)"
      boundingObject Sphere {
        radius 0.05
      }
      physics Physics {
      }
    }
    Solid {
      translation -0.14 -0.25 0
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Sphere {
            radius 0.05
          }
        }
      ]
      name "solid(2)"
      boundingObject Sphere {
        radius 0.05
      }
      physics Physics {
      }
    }
    DEF Car_body Shape {
      appearance Appearance {
        material Material {
        }
        texture ImageTexture {
          url [
            "textures/rock.jpg"
          ]
        }
      }
      geometry Cylinder {
        height 0.5
        radius 0.2
      }
    }
    DEF Left_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 7707.25495982186
        axis 0 0 -1
        anchor 0 -0.15 0.250001
      }
      device [
        RotationalMotor {
          name "left_motor"
        }
      ]
      endPoint Solid {
        translation 0.0010947514963088034 -0.1496954576953517 0.2536600623871694
        rotation 0.24567139809839553 0.6762435613921675 -0.6945071704683958 3.6256377168359735
        children [
          DEF wheel Shape {
            appearance Appearance {
              texture ImageTexture {
                url [
                  "textures/checkered_parquetry.jpg"
                ]
              }
            }
            geometry Cylinder {
              height 0.1
              radius 0.15
            }
          }
        ]
        boundingObject DEF wheel Shape {
          geometry Cylinder {
            height 0.1
            radius 0.15
          }
        }
        physics Physics {
        }
        linearVelocity -0.3538523191117389 -0.0005257481596361841 -0.2716474654483125
        angularVelocity -1.826900231021341 -0.0012107001207215542 2.379749220901711
      }
    }
    DEF Right_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position -2447.7148092169555
        axis 0 0 1
        anchor 0 -0.15 -0.250001
      }
      device [
        RotationalMotor {
          name "right_motor"
        }
      ]
      endPoint Solid {
        translation -3.5590751231605517e-06 -0.14871231060184167 -0.2536138589041783
        rotation 0.16466165084580522 0.7047404990660181 -0.6900922907241567 3.4476001332982524
        children [
          DEF wheel Shape {
            appearance Appearance {
              texture ImageTexture {
                url [
                  "textures/checkered_parquetry.jpg"
                ]
              }
            }
            geometry Cylinder {
              height 0.1
              radius 0.15
            }
          }
        ]
        name "solid(1)"
        boundingObject DEF wheel Shape {
          geometry Cylinder {
            height 0.1
            radius 0.15
          }
        }
        physics Physics {
        }
        linearVelocity -0.3545481393952491 -0.00013959669579656817 -0.27218232836695394
        angularVelocity -1.8268905785323843 -0.0012096400652092222 2.37976095606143
      }
    }
  ]
  boundingObject USE Car_body
  physics Physics {
    mass 0.1
  }
  controller "<extern>"
  linearVelocity -0.35254725675839 0.0008604751903461927 -0.26976622844630205
  angularVelocity 0.005369712848061488 0.0030213325823318996 0.004384677077899702
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 1.82607 -2.35 1.59912
  floorSize 5 5
  wallHeight 2
}
